src/app/user/shared/authentication.service.ts

```diff
@@ -23,11 +29,12 @@ export class AuthService {
       address: [],
       pictureURL: 'assets/images/users/dummy1.jpg'
     }
-    this.isLoggedIn$.next(!!this.currentUser.username)
+    this.isLoggedIn.next(!!this.currentUser);
   }
 
-  isAuthenticated() {
-    return false
+  logoutUser() {
+    this.isLoggedIn.next(false);
+    this.router.navigate(['/user/login']);
   }
 }
 
```

Comment:
(not referencing any code, just commenting an update)
My adjustments are added. I appreciate your in depth feedback! 

I spent a decent bit of time learning more rxjs stuff (observables, subjects, subject behavior, replays, some operators). I was able to learn enough to grasp more understanding and enough to be able to (hopefully successfully) improve my code from before and have it function how it should through better practice.

Reply:
@swhook52